test/e2e/apps/daemon_set.go

```diff
@@ -261,6 +262,17 @@ var _ = SIGDescribe("Daemon set [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred(), "error failing a daemon pod")
 		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, checkRunningOnAllNodes(f, ds))
 		Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to revive")
+
+		By("Wait for the failed daemon pod to be completely deleted.")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, waitFailedDaemonPodDeleted(c, &pod))
+		Expect(err).NotTo(HaveOccurred(), "error waiting for the failed daemon pod to be completely deleted")
+
+		By("Ensure that failed daemonset pod has been removed.")
```

Comment:
We can remove this and below maybe? I think `waitFailedDaemonPodDeleted` already ensures that the pod is deleted.

Reply:
Yes, if the polling is timeout, it will throw an error, so this step is unnecessary.